bundles/org.openhab.binding.gardena/src/main/java/org/openhab/binding/gardena/internal/GardenaSmartWebSocket.java

```diff
@@ -0,0 +1,176 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.gardena.internal;
+
+import java.io.IOException;
+import java.net.URI;
+import java.nio.ByteBuffer;
+import java.time.Instant;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.websocket.api.Session;
+import org.eclipse.jetty.websocket.api.annotations.*;
+import org.eclipse.jetty.websocket.api.extensions.Frame;
+import org.eclipse.jetty.websocket.client.WebSocketClient;
+import org.eclipse.jetty.websocket.common.WebSocketSession;
+import org.eclipse.jetty.websocket.common.frames.PongFrame;
+import org.openhab.binding.gardena.internal.config.GardenaConfig;
+import org.openhab.binding.gardena.internal.model.dto.api.PostOAuth2Response;
+import org.openhab.binding.gardena.internal.model.dto.api.WebSocketCreatedResponse;
+import org.openhab.core.io.net.http.WebSocketFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link GardenaSmartWebSocket} implements the websocket for receiving constant updates from the Gardena smart
+ * system.
+ *
+ * @author Gerhard Riegler - Initial contribution
+ */
+@NonNullByDefault
+@WebSocket
+public class GardenaSmartWebSocket {
+    private final Logger logger = LoggerFactory.getLogger(GardenaSmartWebSocket.class);
+    private final GardenaSmartWebSocketListener socketEventListener;
+    private final long WEBSOCKET_IDLE_TIMEOUT = 300;
+
+    private WebSocketSession session;
+    private WebSocketClient webSocketClient;
+    private boolean closing;
+    private Instant lastPong = Instant.now();
+    private ScheduledExecutorService scheduler;
+    private @Nullable ScheduledFuture<?> connectionTracker;
+    private ByteBuffer pingPayload = ByteBuffer.wrap("ping".getBytes());
```

Comment:
You should specify the charset used here since the system default charset might not be the one you want since it varies from system to system.